src/details/ArborX_DetailsBufferOptimization.hpp

```diff
@@ -221,79 +144,27 @@ void queryImpl(ExecutionSpace const &space, TreeTraversal const &tree_traversal,
   // pre-condition: offset and out are preallocated. If buffer_size > 0, offset
   // is pre-initialized
 
+  using MapType =
+      Kokkos::UnorderedMap<Kokkos::pair<int, int>,
+                           typename OutputView::value_type, ExecutionSpace>;
```

Comment:
I wonder if this is still the right type. Our use is not the usual one, we only do inserts and than a single scan through. Having this work as map is really unnecessary, and I think would lead to more memory jump. I would try to do
```
using MapType = Kokkos::UnorderedMap<Kokkos::Tuple<int, int, typename OutputView::value_type>, void, ExecutionSpace>;
```
(not sure if there is `Kokkos::Tuple`, but you get the gist.